### PR TITLE
COMMERCE-9961 - Product API does not accept expiration date in the past

### PR DESCRIPTION
**Pull request by victorcdp**
Related issue: [COMMERCE-9961](https://issues.liferay.com/browse/COMMERCE-9961)
@brianikim

Problem was that there was a check to see if the product's expiration date was before the current one.
Proposed fix is to remove that and allow any expiration date as long as it is after the product's display date.

### CODE REVIEW

**Comment by victorcdp** [comment]
ci:test:sf

**Comment by victorcdp** [comment]
ci:test:relevant

**Comment by victorcdp** [comment]
ci:test:commerce

**Comment by brianikim** [comment]
@andrea-ale-sbarra , this one needs your review.

**Comment by victorcdp** [comment]
ci:test:relevant

**Comment by victorcdp** [comment]
ci:test:commerce

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java [hunk 1]
```diff
@@ -2508,14 +2508,6 @@ public CPDefinition updateStatus(
 
 		Date modifiedDate = serviceContext.getModifiedDate(date);
 
-		if (status == WorkflowConstants.STATUS_APPROVED) {
-			Date expirationDate = cpDefinition.getExpirationDate();
-
-			if ((expirationDate != null) && expirationDate.before(date)) {
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java:2514]: I don't get why you deleted this block

**Comment by victorcdp** [comment]
@andrea-ale-sbarra  the block you mentioned was doing basically the same as the other one, checking if the new expiration date is before the current date, it's just in another method (updateStatus), and it was being run when updating the product, so the date wasn't getting accepted. Since the ticket is about allowing expiration dates before the current date, it should be removed as well.

**Comment by andrea-ale-sbarra** [comment]
sent here https://github.com/liferay-commerce/liferay-portal/pull/4993

**Pull request closed without merging**